### PR TITLE
librbd: disallow creation of v1 image format

### PR DESCRIPTION
**Pull request by colletj**
V1 image format has been deprecated since the jewel release, so
forbit it. However some internal test cases still need v1 images
so allow their creation if the RBD_FORCE_ALLOW_V1 env var is set.

Signed-off-by: Julien Collet <julien.collet@cern.ch>

### CODE REVIEW

**Comment by colletj** [comment]
@dillaman This is a work-in-progress, is this what you had in mind ?

File: src/librbd/internal.cc [hunk 1]
```diff
@@ -869,7 +869,13 @@ bool compare_by_name(const child_info_t& c1, const child_info_t& c2)
     }
 
     if (old_format) {
-      r = create_v1(io_ctx, image_name.c_str(), size, order);
+      if ( !getenv("RBD_FORCE_ALLOW_V1") ) {
+        lderr(cct) << "Format 1 image creation deprecated. " << dendl;
```
Comment (dillaman) [inline src/librbd/internal.cc]: Nit: ```deprecated``` -> ```unsupported```

File: src/librbd/internal.cc [hunk 2]
```diff
@@ -869,7 +869,13 @@ bool compare_by_name(const child_info_t& c1, const child_info_t& c2)
     }
 
     if (old_format) {
-      r = create_v1(io_ctx, image_name.c_str(), size, order);
+      if ( !getenv("RBD_FORCE_ALLOW_V1") ) {
+        lderr(cct) << "Format 1 image creation unsupported. " << dendl;
+        return -EINVAL;
+      } else {
```
Comment (dillaman) [inline src/librbd/internal.cc]: Nit: you can drop the ```else``` clause since the previous clause will return.

Comment (colletj) [reply src/librbd/internal.cc]: Okay I'll have a look, thanks for pointing this out.

**Comment by colletj** [comment]
Okay I'll have a look, thanks for pointing this out.

File: src/librbd/internal.cc [hunk 3]
```diff
@@ -124,25 +124,25 @@ bool compare_by_name(const child_info_t& c1, const child_info_t& c2)
 } // anonymous namespace
 
   int detect_format(IoCtx &io_ctx, const string &name,
-		    bool *old_format, uint64_t *size)
+                    bool *old_format, uint64_t *size)
```
Comment (dillaman) [inline src/librbd/internal.cc]: Nit: try to avoid changing tabs to spaces -- unless the lines are directly affected by the PR. While I prefer spaces, technically tabs are allowed in the project.

File: src/test/run-rbd-unit-tests.sh [hunk 4]
```diff
@@ -14,4 +14,6 @@ do
     RBD_FEATURES=$i unittest_librbd
 done
 
+unset RBD_FORCE_ALLOW_V1
```
Comment (dillaman) [inline src/test/run-rbd-unit-tests.sh]: Nit: why is this needed?

Comment (colletj) [reply src/test/run-rbd-unit-tests.sh]: It seems it's not -- I fixed it.
Thanks for your feedback

**Comment by dillaman** [comment]
Test failures:
http://qa-proxy.ceph.com/teuthology/jdillaman-2018-03-08_11:44:55-rbd-wip-jd-testing-distro-basic-smithi/2265649/teuthology.log
http://qa-proxy.ceph.com/teuthology/jdillaman-2018-03-08_11:44:55-rbd-wip-jd-testing-distro-basic-smithi/2265656/teuthology.log
http://qa-proxy.ceph.com/teuthology/jdillaman-2018-03-08_11:44:55-rbd-wip-jd-testing-distro-basic-smithi/2265689/teuthology.log

**Comment by dillaman** [comment]
Looks like just one additional test failure:
http://qa-proxy.ceph.com/teuthology/jdillaman-2018-03-15_20:37:28-rbd-wip-jd-testing-distro-basic-smithi/2294312/teuthology.log

**Comment by colletj** [comment]
@dillaman, PR updated. Do you think you could relaunch the tests, please ?

**Comment by dillaman** [comment]
@colletj Still having an issue w/ the cram test -- I think the semi-colon needs to be removed between the environment variable and the ```rbd``` CLI.

http://qa-proxy.ceph.com/teuthology/jdillaman-2018-03-29_21:54:26-rbd-wip-jd-testing-distro-basic-smithi/2334466/teuthology.log

**Comment by colletj** [comment]
@dillaman Thanks for the feedback, I fixed it. Could you give it another try, please ?

**Comment by dillaman** [comment]
Manually merged to master
